fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java

```
@@ -17,62 +17,80 @@
  * 
  * @author Jason W. Thompson -
  *         https://google.com/+JasonWThompson_SoftwareDeveloper
+ * @author David K. Norton 
+ *         david.k.norton@gmail.com
  */
 public class MockPersonDataSource implements PersonDataSource {
     private static final Map<String, Person> PERSONS_BY_UID = new HashMap<>();
     private static final Map<String, Person> DELETED_PERSONS_BY_UID = new HashMap<>();
-    private static final Map<String, Person> PERSON_METADATA_BY_UID = new HashMap<>();
+    private static final Map<String, PersonMetaData> PERSON_METADATA_BY_UID = new HashMap<>();
+    private static final Map<String, PersonMetaData> DELETED_PERSON_METADATA_BY_UID = new HashMap<>();
 
     @Override
     public String createPerson(Person person) {
-        checkNotNull(person, "The parameter person cannot be null.");
+        checkNotNull(person, "The parameter person is null.");
         final String myUUID = UUID.randomUUID().toString();
         SimplePerson myPerson = new SimplePerson();
         myPerson.setUID(myUUID);
-        myPerson.setName(person.getName());
+        myPerson.setName(person.getName());        
+        SimplePersonMetaData metadata = new SimplePersonMetaData(myUUID);
 
         PERSONS_BY_UID.put(myUUID, myPerson);
+        PERSON_METADATA_BY_UID.put(myUUID, metadata);
         return myUUID;
     }
 
     @Override
     public void updatePerson(Person person) {
```

Let's throw an argument exception if the UID of the person doesn't already exist.